spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/servlet/ServletContextInitializerBeans.java

```diff
@@ -318,6 +322,15 @@ private void configureFromAnnotation(ServletRegistrationBean<Servlet> bean, Serv
 			bean.setIgnoreRegistrationFailure(registration.ignoreRegistrationFailure());
 			bean.setLoadOnStartup(registration.loadOnStartup());
 			bean.setUrlMappings(Arrays.asList(registration.urlMappings()));
+
+			for (WebInitParam param : registration.initParameters()) {
+				bean.addInitParameter(param.name(), param.value());
+			}
+
+			MultipartConfig multipart = registration.multipartConfig();
+			bean.setMultipartConfig(new MultipartConfigElement(multipart.location(), multipart.maxFileSize(),
```

Comment:
`MultipartConfigElement` provides a constructor that takes the annotation as an argument.

Reply:
fixed